File: Java(eclipse)/workspace/書籍検索アプリ/src/main/java/todo/model/BookSearchManager.java
package todo.model;

import java.io.IOException;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;

public class BookSearchManager {

	private BookDataManager bookDataManager; //CSV読み込みの管理
	private BookDataSearcher bookDataSearcher; //検索ボタンの動作
	private AllBookDatasDisplay allBookDatasDisplay; //全件表示の動作
	private SearchEngineSearch searchEngineSearch; //検索エンジンでの検索の動作
	private SearchInformationDataManager searchInfoManager;
	
	//------------
	//コンストラクタ
	public BookSearchManager(SearchInformationDataManager searchInfoManager) {
		
		this.searchInfoManager = searchInfoManager;
		initializeParameters();
	}
	
	private void initializeParameters() {
		
		bookDataManager = new BookDataManager();
		bookDataSearcher = new BookDataSearcher(bookDataManager,searchInfoManager);
        searchEngineSearch = new SearchEngineSearch(searchInfoManager);
        allBookDatasDisplay = new AllBookDatasDisplay(bookDataManager,searchInfoManager);
	}
	
	public void loadBookDatas(String filePath) {
		
		bookDataManager.loadBookDatas(filePath);
	}
	
	public void searchBookData(String keyWord,HttpServletRequest request) {
		
		bookDataSearcher.searchBookData(keyWord,request);
	}
	
	public void allBookDataDisplay(HttpServletRequest request) 
		throws ServletException, IOException {
		
		allBookDatasDisplay.allBookDataDisplay(request);
	}
	
	/*public void searchSearchEngine(String target,HttpServletRequest request, 
		HttpServletResponse response)throws ServletException, IOException {
		
		searchEngineSearch.searchSearchEngine(target,request, response);
	}*/
}

File: 1_C#(VisualStudio)/1_workspace_ASP.NETCoreMVC/1_書籍検索アプリ_移植元Java/Models/BookSearchManager.cs
using 書籍検索アプリ_移植元Java.Models.beans;

namespace 書籍検索アプリ_移植元Java.Models {
    public class BookSearchManager {

        private readonly BookDataSearcher bookDataSearcher; //検索ボタンの動作
        private readonly AllBookDatasDisplay allBookDatasDisplay; //全件表示の動作

        //------------
        //コンストラクタ
        public BookSearchManager(BookDataManager bookDataManager) {

            bookDataSearcher = new(bookDataManager);
            allBookDatasDisplay = new(bookDataManager);

            // wwwroot/Resources/bookDataList.csv の絶対パスを取得
            //var filePath = Path.Combine(env.WebRootPath, "Resources", "bookDataList.csv");

            // CSV を読み込む（Java の init() 相当）
            //LoadBookDatas(filePath); // ← Program.csでシングルトン設定済み。ここが1回だけ呼ばれる
        }

        public List<BookDataBean>? SearchBookData(string? keyWord, string? pushedRadio, string[]? chkBoxDataArray) {

            return bookDataSearcher.SearchBookData(keyWord, pushedRadio, chkBoxDataArray);
        }

        public List<BookDataBean>? AllBookDataDisplay(string[] chkBoxDataArray) {

            return allBookDatasDisplay.AllBookDataDisplay(chkBoxDataArray);
        }
    }
}